src/xbot/common/subsystems/BaseSRXControlledSubsystem.java

```diff
@@ -0,0 +1,180 @@
+        
+        masterMotor = factory.getCANTalonSpeedController(masterChannel);
+
+        initializeMotorConfiguration(invertMaster, invertMasterSensor);
+        masterMotor.createTelemetryProperties(systemName + "  master", propManager);
+    }
+    
+    /**
+     * Setup extra values for the masterMotor XCANTalon, and do any necessary setup of your
+     * follower XCANTalon(s).
+     * Some examples:
+     * -Configure your follower XCANTalon(s) to follow masterMotor
+     * -Configure voltage settings, such as configNominalOutputVoltage and configPeakOutputVoltage
+     */
+    protected abstract void setupExtendedMotorProperties();
+    
+    protected void initializeMotorConfiguration(boolean motorInverted, boolean motorSensorInverted) {
+        masterMotor.setFeedbackDevice(FeedbackDevice.QuadEncoder);
+        masterMotor.setInverted(motorInverted);
+        masterMotor.reverseSensor(motorSensorInverted);
+        masterMotor.setControlMode(TalonControlMode.Speed);
+        masterMotor.setProfile(0);
+        
+        setupExtendedMotorProperties();
+    }
+        
+    private void updateMotorPidValues() {
+        masterMotor.setP(pidPropertyManager.getP());
+        masterMotor.setI(pidPropertyManager.getI());
+        masterMotor.setD(pidPropertyManager.getD());
+        masterMotor.setF(pidPropertyManager.getF());
+    }
+    
+    /**
+     * Sets the output power of the system directly - no PID of any kind
+     * @param Power is set to system
+     */
+    public void setPower(double power) {
+        masterMotor.ensureTalonControlMode(TalonControlMode.PercentVbus);
+        masterMotor.set(power);
+    }
+    
+    /**
+     * Returns the last set motor power. Only works if the motor is currently in PercentVbus mode - otherwise, returns 0.
+     * @return Last Motor power
+     */
+    public double getPower() {
+        masterMotor.getOutputVoltage();
+        double inversionFactor = masterMotor.getInverted() ? -1 : 1;
+        return masterMotor.get() * inversionFactor;
+    }
+
+    /**
+     * Gives the system a new speed goal.
+     * @param systemTargetSpeed in Rotations per Second (RPS)
+     */
+    public void setTargetSpeed(double speed) {
+        masterMotor.ensureTalonControlMode(TalonControlMode.Speed);
+        
+        // Update property for dashboard
+        systemTargetSpeed.set(speed);
+        // If there have been any changes to encoder settings or PID setings, apply them.
+        updateMotorPidValues();
+        // Instruct motor about new speed goal
+        masterMotor.set(speed);
+    }
+    
+    public double getSpeed() {
+        return masterMotor.getSpeed();
+    }
+    
+    public boolean isAtSpeed() {
+        return Math.abs(getSpeed() - systemTargetSpeed.get()) <= systemSpeedThresh.get();
+    }
+    
+    public double getTargetSpeed() {
+        return systemTargetSpeed.get();
+    }
+
+    @Override
```

Comment:
do you still do `@Override` for interface methods that aren't shadowing an existing method?